File: frontend/src_working/types/apiTypes/songList.ts
// Song List types matching backend schemas

export interface SongList {
    song_list_id: number;
    admin_user_id: number;
    list_name: string;
    original_filename: string;
    uploaded_at: string;
}

export interface SongListCreate {
    admin_user_id: number;
    list_name: string;
    original_filename: string;
}

export interface SongListUpdate {
    list_name?: string;
    original_filename?: string;
}

File: backend/schemas/song_list.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class SongListBase(BaseModel):
    list_name: str
    original_filename: str


class SongListCreate(SongListBase):
    admin_user_id: int


class SongListUpdate(BaseModel):
    list_name: Optional[str] = None


class SongListResponse(SongListBase):
    song_list_id: int
    admin_user_id: int
    uploaded_at: datetime

    model_config = ConfigDict(from_attributes=True)
